cpp/include/ucxx/inflight_requests.h

```diff
@@ -7,23 +7,38 @@
 #include <map>
 #include <memory>
 #include <mutex>
+#include <utility>
 
 namespace ucxx {
 
 class Request;
 
 typedef std::map<const Request* const, std::shared_ptr<Request>> InflightRequestsMap;
 typedef std::unique_ptr<InflightRequestsMap> InflightRequestsMapPtr;
+typedef std::pair<InflightRequestsMapPtr, InflightRequestsMapPtr> InflightRequestsMapPtrPair;
```

Comment:
**suggestion**: Although "a name is not type safety", this type seems very easy to get confused with. It's a pair of maps, the first of which is (IIUC) "new" requests that we should record as being in flight, while the second is "existing" requests that we should record for cancellation.

This seems easy to misuse (the types of the two things don't differ) and `.first` and `.second` don't give the reader context to know what is wrong. One must "just know" that the first entry is new and the second to cancel.

How about (no type safety, but name safety):

```
struct ChangedRequests {
   InflightRequestsMapPtr new_;
   InflightRequestsMapPtr to_cancel_;
}
```

Reply:
Agreed, done in https://github.com/rapidsai/ucxx/pull/153/commits/2d78accd21d576bd6f426d0bb579ed3179fc370e .